spec.html

```diff
@@ -37807,23 +37818,48 @@ <h1>AtomicLoad ( _typedArray_, _index_ )</h1>
           1. Return GetValueFromBuffer(_buffer_, _indexedPosition_, _elementType_, *true*, ~SeqCst~).
         </emu-alg>
       </emu-clause>
+
+      <emu-clause id="sec-bytelistbitwiseop" aoid="ByteListBitwiseOp">
+        <h1>ByteListBitwiseOp(_op_, _xBytes_, _yBytes_)</h1>
+        <p>The abstract operation ByteListBitwiseOp takes three arguments, a bitwise operator _op_, a List of byte values _xBytes_, and a List of byte values _yBytes_. The operation atomically performs a bitwise operation on all bytes and returns a List of byte values. It performs the following steps:</p>
+        <emu-alg>
+          1. Assert: _xBytes_ and _yBytes_ have equal length.
+          1. Let _result_ be a new empty List.
+          1. Let _i_ be 0.
+          1. Let each element _xByte_ of _xBytes_, do
+            1. Let _yByte_ be _yBytes_[_i_].
+            1. Let _resultByte_ be the result of applying the bitwise operator _op_ to _xByte_ and _yByte_.
+            1. Set _i_ to _i_ + 1.
+            1. Append _resultByte_ to the end of _result_.
+          1. Return _result_.
+        </emu-alg>
+      </emu-clause>
     </emu-clause>
 
     <emu-clause id="sec-atomics.add">
       <h1>Atomics.add ( _typedArray_, _index_, _value_ )</h1>
-      <p>Let `add` denote a semantic function of two List of byte values arguments that applies the addition operation to the Number values corresponding to the List of byte values arguments and returns a List of byte values corresponding to the result of that operation.</p>
       <p>The following steps are taken:</p>
       <emu-alg>
-        1. Return ? AtomicReadModifyWrite(_typedArray_, _index_, _value_, `add`).
+        1. Let _type_ be the Element Type value in <emu-xref href="#table-the-typedarray-constructors"></emu-xref> for _typedArray_.[[TypedArrayName]].
+        1. Let _isLittleEndian_ be the value of the [[LittleEndian]] field of the surrounding agent's Agent Record.
+        1. Let _add_ be a new read-modify-write modification function with parameters (_xBytes_, _yBytes_) that captures _type_ and _isLittleEndian_ and performs the following steps atomically when called:
+          1. Assert: _xBytes_ and _yBytes_ have equal length.
+          1. Let _x_ be RawBytesToNumeric(_type_, _xBytes_, _isLittleEndian_).
+          1. Let _y_ be RawBytesToNumeric(_type_, _yBytes_, _isLittleEndian_).
+          1. Let _T_ be Type(_x_).
+          1. Let _sum_ be _T_::add(_x_, _y_).
+          1. Return NumericToRawBytes(_type_, _sumBytes_, _isLittleEndian_).
```

Comment:
`_sumBytes_` does not appear to be defined. Perhaps you meant `_sum_`?

Also, you maybe need to worry about ensuring you return a list of the same length as was input.

Reply:
Yes I wondered about overflow here, but it is how the rest of TA works (and `|0`, I assume) already, so an assert seems fine here.